extensions/azuremonitor/src/azuremonitorApiFactory.ts

```diff
@@ -0,0 +1,23 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AppContext } from './appContext';
+import { IExtension, AzureMonitorObjectExplorerBrowser } from './azuremonitor';
+import * as constants from './constants';
+import { AzureMonitorObjectExplorerNodeProvider } from './objectExplorerNodeProvider/objectExplorerNodeProvider';
+import * as azdata from 'azdata';
+
+export function createAzureMonitorApi(context: AppContext): IExtension {
+	return {
+		getAzureMonitorObjectExplorerBrowser(): AzureMonitorObjectExplorerBrowser {
+			return {
+				getNode: (explorerContext: azdata.ObjectExplorerContext) => {
+					let oeProvider = context.getService<AzureMonitorObjectExplorerNodeProvider>(constants.ObjectExplorerService);
+					return <any>oeProvider?.findSqlClusterNodeByContext(explorerContext);
```

Comment:
Why are you casting to any here? You should avoid doing that in general for full typing support

Reply:
Looks like `getNode` requires it:
![image](https://user-images.githubusercontent.com/63619224/124665795-06f7da80-de62-11eb-85e2-62edbf4bc631.png)

Reply:
That's saying the type is incorrect, not that it requires any. In this case it seems like at least one issue is that the type you're implementing here says that getNode always returns a value https://github.com/microsoft/azuredatastudio/pull/15397/files/7be34725224c35297915d0986549de6d6a857c0b#diff-f64d59b4291825bbc4f2cbdc23f51c2e020fa6a1c0666db077cfec2ed77ca1daR29 (it's not `Thenable<T> | undefined`)

So you need to make it so that it always returns a promise here, not undefined.